### PR TITLE
Setup cache

### LINKED ISSUE #11: Caching Service
Wrapper around Cache for 
Get 
Set with TTL
Delete
and any other functions that would be needed

### PR DESCRIPTION
**Pull request by arkn98**
Closes #11

### CODE REVIEW

File: config/config.yaml [hunk 1]
```diff
@@ -5,6 +5,11 @@ server:
 logger:
   level: debug
 
+cache:
+  network: tcp
+  addr: 127.0.0.1:6379
```
[the remark this answers is not in the document]
Comment (arkn98) [reply config/config.yaml:10]: Okay sure

File: internal/middleware/middleware.go [hunk 2]
```diff
@@ -31,7 +31,7 @@ func (c *FuncChain) Compose(f http.Handler) http.Handler {
 
 // Append appends the passed-in funcs to the existing middleware chain.
 func (c *FuncChain) Append(funcs ...Func) FuncChain {
-	newChain := make([]Func, 0, len(c.funcs) + len(funcs))
+	newChain := make([]Func, 0, len(c.funcs)+len(funcs))
```
[the remark this answers is not in the document]
Comment (arkn98) [reply internal/middleware/middleware.go:34]: I just ran gofmt

File: internal/config/config.go [hunk 3]
```diff
@@ -19,11 +19,18 @@ func setEnvOverrides() {
...
+
+	// database env overrides
 	_ = viper.BindEnv("database.dbname", utils.EnvDBNameKey)
 	_ = viper.BindEnv("database.user", utils.EnvDBUserKey)
 	_ = viper.BindEnv("database.password", utils.EnvDBPasswordKey)
 	_ = viper.BindEnv("database.host", utils.EnvDBHostKey)
 	_ = viper.BindEnv("database.port", utils.EnvDBPortKey)
+
+	// cache env overrides
+	_ = viper.BindEnv("cache.network", utils.EnvCacheNetworkKey)
```
[the remark this answers is not in the document]
Comment (arkn98) [reply internal/config/config.go:31]: okay sure

File: internal/service/cache/cache.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+		p, e := radix.NewPool(
+			config.GetString("cache.network"),
+			config.GetString("cache.addr"),
+			config.GetInt("cache.connsize"),
+		)
+		if e != nil {
+			err = e
+			return
+		}
+		pool = &Cache{p}
```
[the remark this answers is not in the document]
Comment (arkn98) [reply internal/service/cache/cache.go:38]: No. It will be reachable only when `e` is `nil`.

[the remark this answers is not in the document]
Comment (dyaneshvarun) [reply internal/service/cache/cache.go:39]: Got it, Sorry was confused.

**Pull request merged**